agenta-web/cypress/e2e/testset.cy.ts

```diff
@@ -40,4 +40,187 @@ describe("create a new testset", () => {
                 })
             })
     })
+
+    it("successfully creates the testset and adds options to it", () => {
+        const testsetName = randString(8)
+
+        const countries = [
+            {country: "USA", capital: "Washington DC", dummy: "Stuff"},
+            {country: "Nigeria", capital: "Abuja", dummy: "Thing"},
+            {country: "Egypt", capital: "Cairo", dummy: "Nothing"},
+            {country: "Ethopia", capital: "Addis Ababa", dummy: "Infinity"},
+        ]
+        cy.get(".ag-root-wrapper").as("grid")
+
+        // set values for the cells
+        cy.get("@grid").find('[row-id="0"]').as("row0")
+        cy.get("@grid").find('[row-id="1"]').as("row1")
+        cy.get("@grid").find('[row-id="2"]').as("row2")
+
+        countries.slice(0, 3).forEach(({country, capital}, index) => {
+            cy.get(`@row${index}`).find('[col-id="country"]').click().type(country)
+            cy.get(`@row${index}`).find('[col-id="correct_answer"]').click().type(capital)
+        })
+
+        cy.get('[data-cy="testset-name-input"]').type(testsetName)
+
+        // save and go back
+        cy.get('[data-cy="testset-save-button"]').as("saveButton").click()
+        cy.get(".ant-message-success").should("be.visible")
+
+        cy.get('[data-cy="app-testsets-link"]').as("testsetsLink")
+        cy.clickLinkAndWait("@testsetsLink")
+        cy.url().should("include", "/testsets")
+
+        // validate that the new testset is in the list
```

Comment:
it("When validating that the new testset is in the list", () => {
...
}

You can go ahead now and do it for the rest:)
common syntax is beginning the `it()` description with "When"

Reply:
So, you're saying I should break down the testcase I wrote into smaller testcases?